devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -310,30 +315,40 @@ def configure() {
 	def startValues = zigbee.writeAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT, DataType.INT16, 0x07D0) +
 			zigbee.writeAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT, DataType.INT16, 0x0A28)
 
+	if (isDanfossAlly()) {
+		// setting Min/Max HeatSetPointLimits for Danfoss Ally - MinHeatSetpointLimit: 500 (0x01F4), MaxHeatSetpointLimit: 3500 (0x0DAC)
+		startValues += zigbee.writeAttribute(THERMOSTAT_CLUSTER, MIN_HEAT_SETPOINT_LIMIT, DataType.INT16, 0x01F4) +
+				zigbee.writeAttribute(THERMOSTAT_CLUSTER, MAX_HEAT_SETPOINT_LIMIT, DataType.INT16, 0x0DAC)
+	}
+
 	return binding + startValues + zigbee.batteryConfig() + refresh()
 }
 
-def getBatteryPercentage(rawValue) {
+def getBatteryPercentage(rawValue, attribute) {
 	def result = [:]
 
 	result.name = "battery"
 
-	if (rawValue == 0) {
-		sendEvent(name: "powerSource", value: "mains", descriptionText: "${device.displayName} is connected to mains")
-		result.value = 100
-		result.descriptionText = "${device.displayName} is powered by external source."
-	} else {
-		def volts = rawValue / 10
-		def minVolts = 5
-		def maxVolts = 6.5
-		def pct = (volts - minVolts) / (maxVolts - minVolts)
-		def roundedPct = Math.round(pct * 100)
-		if (roundedPct < 0) {
-			roundedPct = 0
+	if (attribute == BATTERY_VOLTAGE) {
+		if (rawValue == 0) {
+			sendEvent(name: "powerSource", value: "mains", descriptionText: "${device.displayName} is connected to mains")
+			result.value = 100
+			result.descriptionText = "${device.displayName} is powered by external source."
+		} else {
+			def volts = rawValue / 10
+			def minVolts = 5
+			def maxVolts = 6.5
+			def pct = (volts - minVolts) / (maxVolts - minVolts)
+			def roundedPct = Math.round(pct * 100)
+			if (roundedPct < 0) {
+				roundedPct = 0
+			}
+			result.value = Math.min(100, roundedPct)
 		}
-		result.value = Math.min(100, roundedPct)
-		result.descriptionText = "${device.displayName} battery has ${result.value}%"
+	} else if (attribute == BATTERY_PERCENTAGE_REMAINING) {
+		result.value = Math.min(100, rawValue)
 	}
+	result.descriptionText = "${device.displayName} battery has ${result?.value}%"
```

Comment:
It is probably best to just accept the default for the capability so that things are more uniform. I know that it was in the original code.

Reply:
> It is probably best to just accept the default for the capability so that things are more uniform. I know that it was in the original code.

@dkirker What did You mean by that? What's the default for the capability in that case ?

Reply:
I think the means not putting any description text, since I think it will be generated automatically by the capability.

Reply:
@dkirker ?

Reply:
@KKlimczukS Let's just remove the ones for battery for now (there are two). We have translations in the capability definitions, and I think the event history service that the new client uses only looks at that right now anyway.

Reply:
@dkirker ok, done.